java/src/main/java/com/luminoso/pack64/Pack64.java

```diff
@@ -0,0 +1,128 @@
+package com.luminoso.pack64;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * pack64.pack: encodes a vector into a pack64'd string.
+ * pack64.unpack: decodes a pack64'd string into a vector.
+ *
+ * This library is for decoding a packed vector format, defined originally
+ * in the Python package `pack64`.
+ * 
+ * The format uses URL-safe base64 to encode an exponent followed by several
+ * 18-bit signed integers.
+ *
+ * @copyright
+ * (c) Copyright 2019 Luminoso Technologies, Inc
+ * 
+ * @license
+ * MIT License - See LICENSE file
+ * 
+ */
+
+public class Pack64 {
+
+    // 2^17 is the number that makes an 18-bit signed integer go negative.
+    static int SIGN_BIT = 131072;
+    static Double ROUND_MARGIN = SIGN_BIT / (SIGN_BIT - 0.5);
+
+    static char[] alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_".toCharArray();
+    private static final Map<Character,Integer> alphabet_map;
+    static {
+        Map<Character, Integer> aMap = new HashMap<Character,Integer>();
+        int i;
+        for (i=0;i<alphabet.length;i++) {
```

Comment:
For consistency, add spaces around the operators, as you do in line 57

Reply:
fixed